eodag/rest/utils.py

```diff
@@ -1046,6 +1050,126 @@ def fetch_collection_queryable_properties(
     return queryable_properties
 
 
+def add_provider_queryables(provider: str, queryables: dict):
+    """Add the queryables fetched from the given provider to the default queryables.
+    If the queryables endpoint is not supported by the provider, the original queryables are returned
+    :param provider: The provider.
+    :type provider: str
+    :param queryables: default queryables to which provider queryables will be added
+    :type queryables: dict
+    :returns queryable_properties: A dict containing the formatted queryable properties
+                                   including queryables fetched from the provider.
+    :rtype dict
+    """
+    search_plugin = next(
+        eodag_api._plugins_manager.get_search_plugins(provider=provider)
+    )
+    search_type = search_plugin.config.type
+    if search_type == "StacSearch":
+        queryables_url = search_plugin.config.api_endpoint.replace(
+            "/search", "/queryables"
+        )
+    else:
+        queryables_url = getattr(search_plugin.config, "queryables_endpoint", "")
```

Comment:
I do not think we should handle non-stac queryables (using `queryables_endpoint`) as we will then need to define how to read each provider-specific queryables

Reply:
changed